probe-rs/src/architecture/arm/dp/register_generation.rs

```diff
@@ -13,7 +13,7 @@ macro_rules! define_dp_register {
     )
     => {
         $(#[$outer])*
-        #[allow(non_snake_case)]
+        #[expect(non_snake_case)]
```

Comment:
I don't think we should do this here. This depends on what is used in the macro, so this lint doesn't necessarily trigger.

Or is there something which always causes this?

Reply:
No, I just search-and-replaced and missed this spot when restoring macros. I'll restore this a bit later, along with your other observations.